### ISSUE TITLE
C1-W2 dataset issue

### ISSUE THREAD

**Issue by mertcanozdemir** [amanchadha/coursera-deep-learning-specialization#63]
The dataset in C1-W2 is not included in raw file

**Comment by AayushSameerShah** [comment]
Yeah, it gives the following:

```
Cloning into 'E:\GitHub Projects\coursera-deep-learning-specialization'...
remote: Enumerating objects: 1530, done.        
remote: Counting objects: 100% (307/307), done.        
remote: Compressing objects: 100% (171/171), done.        
remote: Total 1530 (delta 173), reused 136 (delta 136), pack-reused 1223 (from 3)        
Receiving objects: 100% (1530/1530), 209.86 MiB | 6.19 MiB/s, done.
Resolving deltas: 100% (261/261), done.
Updating files: 100% (1055/1055), done.
Downloading C1 - Neural Networks and Deep Learning/Week 2/Logistic Regression as a Neural Network/datasets/test_catvnoncat.h5 (617 KB)
Error downloading object: C1 - Neural Networks and Deep Learning/Week 2/Logistic Regression as a Neural Network/datasets/test_catvnoncat.h5 (6a9d39c): Smudge error: Error downloading C1 - Neural Networks and Deep Learning/Week 2/Logistic Regression as a Neural Network/datasets/test_catvnoncat.h5 (6a9d39ce52fa85953c5834695b8534a30ed8cfd54971fb2f080b5a24d002b951): batch response: This repository is over its data quota. Account responsible for LFS bandwidth should purchase more data packs to restore access.

Errors logged to 'E:\GitHub Projects\coursera-deep-learning-specialization\.git\lfs\logs\20250212T071433.379082.log'.
Use `git lfs logs last` to view the log.
error: external filter 'git-lfs filter-process' failed
fatal: C1 - Neural Networks and Deep Learning/Week 2/Logistic Regression as a Neural Network/datasets/test_catvnoncat.h5: smudge filter lfs failed
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/'


Would you like to retry cloning ?
```

**Which is:**

This error means that the repository you are trying to clone uses **Git LFS (Large File Storage)** for some files, and the account that owns the repo has **exceeded its LFS bandwidth quota**. Specifically, the file `test_catvnoncat.h5` (a dataset file) could not be downloaded because the Git LFS quota for the repository has run out.

### How to Fix It:

#### 1. **Check if you actually need those LFS files**  
Since cloning was successful but checkout failed, you can still inspect the repo with:
   ```
   git status
   ```
   If you don’t need the `.h5` files, you can manually work with the rest of the code.

#### 2. **Disable LFS Download (If Not Needed)**
   ```
   GIT_LFS_SKIP_SMUDGE=1 git clone <repo_url>
   ```
   This will clone the repository without downloading LFS files.

#### 3. **Manually Download the Missing File**
   - Sometimes, the dataset is available elsewhere (like Coursera itself).
   - Check the repository’s README or issues for an alternative link.
   - If it's a public repo, the owner might have shared a direct download link.

#### 4. **Use a Different GitHub Account (If You Own the Repo)**
   If this is **your** repository and you own the account, you'll need to **buy additional LFS storage** from GitHub or **delete and re-upload large files** outside of LFS.

#### 5. **Retry Later**  
   If the repo owner resets their quota (GitHub refreshes LFS limits monthly), you might be able to download it later.

### TL;DR:  
The file is stuck because of GitHub LFS quota limits. Either:
- Skip LFS downloads (`GIT_LFS_SKIP_SMUDGE=1`)
- Download the missing file manually
- Wait for the repo owner to fix it

Let me know if you need help finding the dataset manually.
